reactionimp/icemachinesplus
# Fix for "Cart container has its class changed on first load"

Resolves #121  
Impact: **major**  
Type: **bugfix**

## Issue
When the template is hydrated, the  CartIcon somehow picks up a wrong parent. Not sure what the exact reason is, but I assume it has to do with

react-addons-create-fragment package. I couldn't verify it with 100% certainty, because whenever I tried to use the actual response of `HTTP.get(`http://67.205.163.60/getprerenderdata?pagepath=${targetURL}`);`, the template was out-of-sync with the VirtualDOM.

It worked for me when using a dummy response where I put some literal HTML into custom/server/customRoutesManagement/main.js.

## Solution
  Remove react-addons-create-fragment from code base as it seems to make
  problems when used in combination with ReactDOM.hydrate() / SSR

  Replace it with React.Fragment , which is available since React 16.2


## Breaking changes
none

## Testing
1. Ensure that PreRender.io returns the newest HTML
2. Navigate to landing page
3. After template hydration, the cart icon should visually remain unchanged.
